extensions/markdown-language-features/src/languageFeatures/copyFiles/shared.ts

```diff
@@ -56,9 +56,36 @@ export const mediaMimes = new Set([
 	'audio/x-wav',
 ]);
 
+export async function getMarkdownLink(document: vscode.TextDocument, ranges: readonly vscode.Range[], dataTransfer: vscode.DataTransfer, token: vscode.CancellationToken): Promise<{ additionalEdits: vscode.WorkspaceEdit; label: string } | undefined> {
+	if (ranges.length === 0) {
+		return;
+	}
+
+	const selectedTexts = ranges.map(range => document.getText(range));
```

Comment:
You should be able to call `getText` directly in the loop below instead of creating this array